include/shad/core/impl/modifyng_sequence_ops.h

```diff
@@ -28,13 +28,289 @@
+           ForwardIt2* res_ptr) {
+          auto d_first_ = std::get<2>(args);
+          auto op = std::get<3>(args);
+          auto local_range =
+              itr_traits::local_range(std::get<0>(args), std::get<1>(args));
+          auto begin = local_range.begin();
+          auto end = local_range.end();
+          *res_ptr = std::transform(begin, end, d_first_, op);
+        },
+        std::make_tuple(first1, last1, res, unary_op), &res);
+  }
+  return res;
+}
+
+template <typename ForwardIt, typename Generator>
+void generate(distributed_parallel_tag&& policy, ForwardIt first,
+              ForwardIt last, Generator generator) {
+  using T = typename ForwardIt::value_type;
+  using itr_traits = distributed_iterator_traits<ForwardIt>;
+  auto localities = itr_traits::localities(first, last);
+
+  rt::Handle H;
+  for (auto locality = localities.begin(), end = localities.end();
+       locality != end; ++locality) {
+    rt::asyncExecuteAt(
+        H, locality,
+        [](rt::Handle&,
+           const std::tuple<ForwardIt, ForwardIt, Generator>& args) {
+          auto begin = std::get<0>(args);
+          auto end = std::get<1>(args);
+          auto generator = std::get<2>(args);
+          auto local_range = itr_traits::local_range(begin, end);
+          auto lbegin = local_range.begin();
+          auto lend = local_range.end();
+
+          // call the generator to align with the offset
+          auto it = itr_traits::iterator_from_local(begin, end, lbegin);
+          for (auto calls = std::distance(begin, it); calls; --calls)
+            auto tmp = generator();
+
+          std::generate(lbegin, lend, generator);
+        },
+        std::make_tuple(first, last, generator));
+  }
+
+  rt::waitForCompletion(H);
+}
+
+template <typename ForwardIt, typename Generator>
+void generate(distributed_sequential_tag&& policy, ForwardIt first,
+              ForwardIt last, Generator generator) {
+  using T = typename ForwardIt::value_type;
+  using itr_traits = distributed_iterator_traits<ForwardIt>;
+  auto localities = itr_traits::localities(first, last);
+
+  for (auto locality = localities.begin(), end = localities.end();
+       locality != end; ++locality) {
+    rt::executeAt(locality,
+                  [](const std::tuple<ForwardIt, ForwardIt, Generator>& args) {
+                    auto begin = std::get<0>(args);
+                    auto end = std::get<1>(args);
+                    auto generator = std::get<2>(args);
+                    auto local_range = itr_traits::local_range(begin, end);
+                    auto lbegin = local_range.begin();
+                    auto lend = local_range.end();
+
+                    // call the generator to align with the offset
+                    auto it =
+                        itr_traits::iterator_from_local(begin, end, lbegin);
+                    for (auto calls = std::distance(begin, it); calls; --calls)
+                      auto tmp = generator();
```

Comment:
tmp is unused and not needed.  You want that loop to be executed only if generate() has side-effects regardless of the return value.

Reply:
I agree.